OpenRA.Mods.Common/Traits/Player/ProvidesPrerequisite.cs

```diff
@@ -64,19 +63,37 @@ public IEnumerable<string> ProvidesPrerequisites
 
 		public void OnOwnerChanged(Actor self, Player oldOwner, Player newOwner)
 		{
-			if (info.ResetOnOwnerChange)
+			if (Info.ResetOnOwnerChange)
 				Update(newOwner, newOwner.Faction.InternalName);
 		}
 
 		void Update(Player owner, string faction)
 		{
+			if (IsTraitDisabled)
+			{
+				enabled = false;
+				return;
+			}
+
 			enabled = true;
 
-			if (info.Factions.Any())
-				enabled = info.Factions.Contains(faction);
+			if (Info.Factions.Any())
+				enabled = Info.Factions.Contains(faction);
+
+			if (Info.RequiresPrerequisites.Any() && enabled)
+				enabled = owner.PlayerActor.Trait<TechTree>().HasPrerequisites(Info.RequiresPrerequisites);
```

Comment:
Please cache the `TechTree` trait, you can compare with how `PlayerResources` is handled in e.g. the `CashTricker` trait.